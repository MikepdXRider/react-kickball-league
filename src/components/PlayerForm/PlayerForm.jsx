import React from 'react'

export default function PlayerForm({edit = false, playerNameStr, setPlayerNameStr, playerPositionStr, setPlayerPositionStr, submitFn, teamDataArr, playerTeamIdNum, setPlayerTeamIdNum}) {
    return (
        <form onSubmit={(e) => submitFn(e)}>
            <fieldset>
                <legend>{edit ? 'Edit Player' : 'Add Player'}</legend>

                <label htmlFor="name">Name: </label>
                <input type="text" id="name" value={playerNameStr} onChange={(e) => setPlayerNameStr(e.target.value)} name="Player-name" required/>

                <label htmlFor="position">Position: </label>
                <input type="text" id="position" value={playerPositionStr} onChange={(e) => setPlayerPositionStr(e.target.value)} name='Player-city' required/>

                {
                    teamDataArr 
                    && <>
                        <label htmlFor="team">Team: </label>
                        <select value={playerTeamIdNum} onChange={(e) => setPlayerTeamIdNum(e.target.value)} name="teams" id="team" required>
                        <option value=''>-</option>
                        {teamDataArr.map(teamsDataObj => <option key={teamsDataObj.id} name={teamsDataObj.name} value={teamsDataObj.id}>{teamsDataObj.name}</option>)}
                        </select>
                    </>
                }

                <input type="submit" />
            </fieldset>
        </form>
    )
}
