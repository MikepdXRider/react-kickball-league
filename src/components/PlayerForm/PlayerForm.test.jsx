import { render, screen } from "@testing-library/react"
import PlayerForm from "./PlayerForm.jsx"


const testProps = {
    edit: false,
    playerNameStr: 'test-name',
    playerPositionStr: 'test-state',
    playerTeamIdNum: 1,
    setPlayerTeamIdNum: 'test-fn',
    setPlayerNameStr: 'test-fn',
    setPlayerPlayerStr: 'test-fn',
    submitFn: 'test-fn',
    teamDataArr: [{id: 1, name: 'test-team'}]
}

it('renders PlayerForm component', () =>{
    const { container } = render(<PlayerForm {...testProps} />)
    const text = screen.getByText(/add player/i);

    expect(container).toMatchSnapshot();
    expect(text).toBeInTheDocument();
})