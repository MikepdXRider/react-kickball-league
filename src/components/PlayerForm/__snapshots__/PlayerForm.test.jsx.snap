// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayerForm component 1`] = `
<div>
  <form>
    <fieldset>
      <legend>
        Add Player
      </legend>
      <label
        for="name"
      >
        Name: 
      </label>
      <input
        id="name"
        name="Player-name"
        required=""
        type="text"
        value="test-name"
      />
      <label
        for="position"
      >
        Position: 
      </label>
      <input
        id="position"
        name="Player-city"
        required=""
        type="text"
        value="test-state"
      />
      <label
        for="team"
      >
        Team: 
      </label>
      <select
        id="team"
        name="teams"
        required=""
      >
        <option
          value=""
        >
          -
        </option>
        <option
          name="test-team"
          value="1"
        >
          test-team
        </option>
      </select>
      <input
        type="submit"
      />
    </fieldset>
  </form>
</div>
`;
